### PR TITLE
doc/stdenv: document meson variables

### PR DESCRIPTION
**Pull request by worldofpeace**
###### Motivation for this change
Before:
Screenshot from 2019-08-05 21 56 45

After:
Screenshot from 2019-08-05 21 55 11

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

:sparkling_heart: For @grahamc for helping me in IRC really quickly.

### CODE REVIEW

File: doc/stdenv.xml [hunk 1]
```diff
@@ -2755,17 +2755,88 @@ nativeBuildInputs = [ breakpointHook ];
...
+         </para>
+        </listitem>
+       </varlistentry>
+       <varlistentry>
+        <term>
+         <varname>crossMesonFlags</varname>
+        </term>
+        <listitem>
+         <para>
+          Controls the flags passed to Meson for cross compilation.
```
Comment (worldofpeace) [inline doc/stdenv.xml]: I don't actually see this being used in nixpkgs ever...

**Comment by vcunat** [comment]
Oh, there's `mesonBuildType` :-)  I've been doing this via the usual `mesonFlags`, though I suppose it can be slightly confusing when the same option is supplied multiple times.

**Comment by worldofpeace** [comment]
uhh, auto formatting is soo annoying.

**Comment by worldofpeace** [comment]
backported in c8c01e2a9566008055a6374e5795f90c3b090898
 without auto formatting.

**Pull request merged**